### DISCUSSION TITLE
Changing batch_size after Pipeline initialization?

### DISCUSSION THREAD

**Discussion by asusdisciple** [pyannote/pyannote-audio#1610] [Q&A]
So I know this is a very basic question, but I am not sure how I can change the `batch_size` - and parameters in general for pipelines and the models, which are used in the pipeline AFTER it is initialized! Is there a tutorial on this? I know how to load them from a `.yaml` with:

```
pipe =  Pipeline.from_pretrained("pyannote/speaker-diarization-3.1",
                                                          use_auth_token="xxxx",
                                                          cache_dir="models",
                                                          hparams_file="models/config.yaml")
```

However I found it very hard to change parameters after this because `pipe.parameters()` only gets the current configuration. Also when I instantiate a pipeline I found this code in the tutorial:

```
finetuned_pipeline.instantiate({
    "segmentation": {
        "threshold": best_segmentation_threshold,
        "min_duration_off": 0.0,
    },
    "clustering": {
        "method": "centroid",
        "min_cluster_size": 15,
        "threshold": best_clustering_threshold,
    },
})
```

However is there any documentation where I can find a list of usable hyperparameters? Like a big dummy dictionary, which tells me how to set them? In the case above for example I do not know how to set the `batch_size`. Do I Have to add a new key "Pipeline" with this parameter nested in another dict? I also tried adding `batch_size_segmentation` as in the `config.yaml` in the segmentation dict but it did not work.

Any help would be appreciated :)

**Comment by hbredin** [comment]
`pipeline.segmentation_batch_size = 32` should do the trick.

**Reply by asusdisciple** [reply]
Thanks a lot! This did the trick indeed.
